### PR TITLE
Stackedetcd e2e

### PR DESCRIPTION
**Pull request by panktishah26**
*Issue #, if available:* [#229](https://github.com/aws/eks-anywhere/issues/229)

*Description of changes:*

E2E: Removed duplicate e2e tests and added tests for stacked etcd topology

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/304#" title="Author self-approved">panktishah26</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [panktishah26]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: test/e2e/stackedetcd_test.go
Comment (mrajashree) [inline test/e2e/stackedetcd_test.go:23]: I think let's keep the cp count 3 here and for test `TestVSphereKubernetes120StackedEtcdUbuntu` too and we can remove test `TestVSphereKubernetes120StackedEtcdBottlerocket3CP2Workers` since it'll get covered by this test

File: test/e2e/stackedetcd_test.go
Comment (mrajashree) [inline test/e2e/stackedetcd_test.go:28]: can we also add cluster creation tests for 1.21

**Review by mrajashree — APPROVED** [review]
/lgtm

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/304#pullrequestreview-768239328" title="LGTM">mrajashree</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/304#" title="Author self-approved">panktishah26</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [mrajashree,panktishah26]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
